components/library/webkitgtk/patches/13-solaris-support.patch

```diff
@@ -134,3 +154,58 @@ index 6577423..b882b9e 100644
  #else
      return "unknown"_s;
  #endif
+--- webkitgtk-2.42.5/Source/JavaScriptCore/runtime/MachineContext.h.~1~	2023-09-19 04:27:45.475782200 -0400
++++ webkitgtk-2.42.5/Source/JavaScriptCore/runtime/MachineContext.h	2024-02-18 01:24:47.690986457 -0500
+@@ -33,6 +33,7 @@
+ #include <wtf/PlatformRegisters.h>
+ #include <wtf/PointerPreparations.h>
+ #include <wtf/StdLibExtras.h>
++#include <sys/regset.h>
+ 
+ namespace JSC {
+ namespace MachineContext {
+@@ -193,7 +194,7 @@
+ #error Unknown Architecture
+ #endif
+ 
+-#elif OS(FUCHSIA) || OS(LINUX)
++#elif OS(FUCHSIA) || OS(LINUX) || OS(SOLARIS)
+ 
+ #if CPU(X86)
+     return reinterpret_cast<void*&>((uintptr_t&) machineContext.gregs[REG_ESP]);
+@@ -338,7 +339,7 @@
+ #error Unknown Architecture
+ #endif
+ 
+-#elif OS(FUCHSIA) || OS(LINUX)
++#elif OS(FUCHSIA) || OS(LINUX) || OS(SOLARIS)
```

Comment:
Are you sure these changes are okay?  The comment below mentions glibc...

Reply:
The walkcontext(3C) man page states that the three backtrace() related functions were added for glibc compatibility.  I am unable to confirm if backtrace actually works in webkitgtk, but this patch is implemented on at least one Solaris build without ill effect.

Reply:
How is `backtrace()` and/or `walkcontext()` related to this file?  I do not see any reference to them in the file.

The comment mentions `glibc's sys/ucontext.h`.  Maybe our `sys/ucontext.h` is compatible, but if so I'd appreciate some comment stating so.

I'm not sure what exactly you mean by _one Solaris build_, but apparently something does not work properly since you stated that webkit crashes.

Thank you.

Reply:
I do not have the background to compare sys/ucontext.h.  Please reject this PR if you don't like it.

Reply:
> I do not have the background to compare sys/ucontext.h. Please reject this PR if you don't like it.

It is not that I do not like the PR.  I [already said](https://github.com/OpenIndiana/oi-userland/pull/16211#issuecomment-1960459918) it is an improvement of the current situation.  I'm just asking to understand what's going on,

And, BTW, I do not have rights to reject PRs :-).